distatis_weights_upgma.R

```diff
@@ -0,0 +1,114 @@
+#Comp Bio Final Project
+#Helen Pennington
+#aim 3
+
+#Packages
+library(DistatisR)
+library(data.table)
+library(phangorn)
+library(pvclust)
+library(ape)
+
+#qsub inputs
+args = commandArgs(trailingOnly=TRUE)
+print(args)
+seq <- ''
+structure <- ''
+sample <- ''
+
+# test if there is at least 3 arguments: if not, return an error
+if (length(args) < 3) {
+  stop('At least three arguments must be supplied.', call.=FALSE)
+} else if (length(args)==3) {
+  seq = args[1] 
+  structure = args[2] 
+  sample = args[3]
+}
+
+print(seq)
+print(structure)
+print(sample)
+
+#sequence distance matrix
+seq <- fread(seq)
+rownames <- seq$V1
+seq <- seq[,-1]
+seq <- as.matrix(seq)
+rownames(seq) <- rownames
+head(seq)
+
+#structure distance matrix
+structure <- fread(structure)
+rownames <- structure$V1
+structure <- structure[,-1]
+structure <- as.matrix(structure)
+rownames(structure) <- rownames
+head(structure)
+
+#Combine into 3D matrix
+combined_matrix <- array(c(seq, structure), c(21, 21, 2), dimnames =list(rownames,rownames))
+#run distatis
+distatis_output <-distatis(combined_matrix, Norm ="MFA", Distance = TRUE, RV = TRUE, nfact2keep = 2, compact = FALSE)
+#distatis outputs needed
+#pull out combined "compromise" matrix
+combined_distatis_matrix <- distatis_output$res4Splus$Splus
+#check which weights were used
+combined_distatis_weights <- distatis_output$res4Splus$alphaWeights
+combined_distatis_weights
+
+#UPGMA
+phylogeneticTree <- function(tree,name,newick){
+  #write tree as Newick file for scoring
+  ape::write.tree(tree, file=paste0('/projectnb/bi720/MMG/compbio/comparisons/',name,'.txt'))
+  #save visual of tree
+  png(filename = paste0('/projectnb/bi720/MMG/compbio/Aim3_trees/',name,'.png'))
```

Comment:
would be nice to make this either a relative filepath or have it be determined by  command-line argument